### PR TITLE
Add spill stats utilities

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Add spill runs stats plus some utilities used by Prestissimo

### CODE REVIEW

File: velox/exec/Spill.h [hunk 1]
```diff
@@ -143,6 +143,8 @@ class SpillFile {
 
 /// Provides the fine-grained spill execution stats.
 struct SpillStats {
+  /// The number of times that spilling runs on an operator.
+  uint64_t spillRuns{0};
```
[the remark this answers is not in the document]
Comment (xiaoxmeng) [reply velox/exec/Spill.h:147]: I think spillRuns is sufficient?

File: velox/exec/Spill.cpp [hunk 2]
```diff
@@ -449,33 +452,54 @@ SpillStats& SpillStats::operator+=(const SpillStats& other) {
...
+  return (spillRuns >= other.spillRuns) &&
+      (spilledBytes >= other.spilledBytes) &&
+      (spilledRows >= other.spilledRows) &&
+      (spilledPartitions >= other.spilledPartitions) &&
+      (spilledFiles >= other.spilledFiles) &&
+      (spillFillTimeUs >= other.spillFillTimeUs) &&
+      (spillSortTimeUs >= other.spillSortTimeUs) &&
+      (spillSerializationTimeUs >= other.spillSerializationTimeUs) &&
+      (spillDiskWrites >= other.spillDiskWrites) &&
+      (spillFlushTimeUs >= other.spillFlushTimeUs) &&
```
[the remark this answers is not in the document]
Comment (xiaoxmeng) [reply velox/exec/Spill.cpp:496]: I am using >= in VELOX_GE for check. I can disable the reverse operator

File: velox/exec/Spill.cpp [hunk 3]
```diff
@@ -449,33 +452,74 @@ SpillStats& SpillStats::operator+=(const SpillStats& other) {
...
+  UPDATE_COUNTER(spillFlushTimeUs);
+  UPDATE_COUNTER(spillWriteTimeUs);
+#undef UPDATE_COUNTER
+  VELOX_CHECK_LE(
+      !!(gtCount > 0) + !!(ltCount > 0),
+      1,
+      "gtCount {} ltCount {}",
+      gtCount,
+      ltCount);
+  return ltCount > 0;
```
Comment (tanjialiang) [inline velox/exec/Spill.cpp:505]: VELOX_CHECK(!(gtCount > 0 && ltCount > 0))?

File: velox/exec/Spill.cpp [hunk 4]
```diff
@@ -449,33 +452,74 @@ SpillStats& SpillStats::operator+=(const SpillStats& other) {
...
+}
+
+bool SpillStats::operator>(const SpillStats& other) const {
+  return !(*this < other) && (*this != other);
+}
+
+bool SpillStats::operator>=(const SpillStats& other) const {
+  return (*this > other) || (*this == other);
```
Comment (tanjialiang) [inline velox/exec/Spill.cpp:513]: !(*this < other)

File: velox/exec/Spill.cpp [hunk 5]
```diff
@@ -449,33 +452,74 @@ SpillStats& SpillStats::operator+=(const SpillStats& other) {
...
+}
+
+bool SpillStats::operator<=(const SpillStats& other) const {
+  return (*this < other) || (*this == other);
```
Comment (tanjialiang) [inline velox/exec/Spill.cpp:517]: !(*this > other)

File: velox/exec/tests/SpillTest.cpp [hunk 6]
```diff
@@ -679,3 +680,69 @@ INSTANTIATE_TEST_SUITE_P(
...
+  VELOX_ASSERT_THROW(stats1 <= stats2, "");
+  VELOX_ASSERT_THROW(stats1 >= stats2, "");
+  ASSERT_NE(stats1, stats2);
+  const SpillStats zeroStats;
+  stats1.reset();
+  ASSERT_EQ(zeroStats, stats1);
+  ASSERT_EQ(
+      stats2.toString(),
+      "spillRuns[100] spilledBytes[1.00KB] spilledRows[1031] spilledPartitions[1025] spilledFiles[1026] spillFillTimeUs[1.03ms] spillSortTime[1.03ms] spillSerializationTime[1.03ms] spillDiskWrites[1028] spillFlushTime[1.03ms] spillWriteTime[1.03ms]");
+}
```
Comment (tanjialiang) [inline velox/exec/tests/SpillTest.cpp:748]: Do you think we can add some logic check? like:
a = ...
b = ...
assert_true(a < b)
assert_true(a <= b)
assert_false(a >= b)
assert_false(a > b)
assert_false(a == b)
assert_true(a != b)

**Comment by tanjialiang** [comment]
Thanks @xiaoxmeng for the change! Some minor comments

File: velox/exec/Spill.cpp (hunk 3 above)
Comment (xiaoxmeng) [reply velox/exec/Spill.cpp:505]: Both can be zero

**Review by tanjialiang — APPROVED** [review]

File: velox/exec/Spill.cpp (hunk 3 above)
Comment (xiaoxmeng) [reply velox/exec/Spill.cpp:505]: Yeah, simpler. Update.

File: velox/exec/tests/SpillTest.cpp (hunk 6 above)
Comment (xiaoxmeng) [reply velox/exec/tests/SpillTest.cpp:758]: Good point!

**Pull request closed without merging**